### PR TITLE
feat(dashmate): verify system requirements

### PR DESCRIPTION
**Pull request by shumkov**
## Issue being fixed or feature implemented

During a node setup, dashmate should verify and suggest minimal system requirements to users 

## What was done?

- Verify minimal system requirements in `dashmate setup` command: CPU, RAM, free disk space 

## How Has This Been Tested?

Running `dashmate setup` locally 

## Breaking Changes

None

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [x] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/dashmate/src/listr/tasks/setup/regular/verifySystemRequirementsTaskFactory.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+   * @returns {Listr}
+   */
+  async function verifySystemRequirementsTask() {
+    return new Listr([
+      {
+        title: 'System requirements',
+        task: async (ctx, task) => {
+          await dockerCompose.throwErrorIfNotInstalled();
+
+          const MINIMUM_CPU_CORES = ctx.isHP ? 4 : 2;
```
Comment (shumkov) [inline packages/dashmate/src/listr/tasks/setup/regular/verifySystemRequirementsTaskFactory.js:24]: The minimal for MN is 2 CPUs, 60 GB of disk space and 4GB ram + 2GB of swap possibly

I guess for full node should be less

File: packages/dashmate/src/listr/tasks/setup/regular/verifySystemRequirementsTaskFactory.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+          const lessCpuFrequency = hostCpuCores
+            .find((core) => (core.speed / 1000) < MINIMUM_CPU_FREQUENCY);
+
+          if (lessCpuFrequency) {
+            const cpuFrequency = (lessCpuFrequency.speed / 1000); // Convert to GHz
+
+            warnings.push(`${cpuFrequency.toFixed(1)}GHz CPU frequency. Minimum required is ${MINIMUM_CPU_FREQUENCY}GHz`);
+          }
+
+          // Check RAM
```
Comment (shumkov) [inline packages/dashmate/src/listr/tasks/setup/regular/verifySystemRequirementsTaskFactory.js:53]: Shall we check swap?
```sh
npm install systeminformation
```
```js
        // Get swap information
        const swap = await si.mem();
        const swapTotalGb = (swap.swaptotal / (1024 ** 3)).toFixed(2); // Convert bytes to GB and format to 2 decimal places
        const swapUsedGb = (swap.swapused / (1024 ** 3)).toFixed(2); // Convert bytes to GB and format to 2 decimal places
```

File: packages/dashmate/src/listr/tasks/setup/regular/verifySystemRequirementsTaskFactory.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+          if (warnings.length > 0) {
+            const warningsText = warnings.map((warning) => `    - ${warning}`).join('\n');
+
+            const header = chalk`  Minimal requirements aren't met:
+
+{red ${warningsText}}
+
+    Dash Platform needs more minerals`;
+            // TODO: Write some text here
```
Comment (thephez) [inline packages/dashmate/src/listr/tasks/setup/regular/verifySystemRequirementsTaskFactory.js:99]: Not sure what we want to say here. It looks like we will allow proceeding even without meeting the requirements?
```suggestion
    Dash Platform needs more resources than the current system provides. Upgrading system resources to meet the minimum requirements is recommended before proceeding.`;
```

File: packages/dashmate/src/listr/tasks/setup/regular/verifySystemRequirementsTaskFactory.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+              type: 'toggle',
+              header,
+              message: 'Are you sure you want to proceed?',
+              enabled: 'Yes',
+              disabled: 'No',
+              initial: false,
+            });
+
+            if (!proceed) {
+              throw new Error('System requirements are not met');
```
Comment (thephez) [inline packages/dashmate/src/listr/tasks/setup/regular/verifySystemRequirementsTaskFactory.js:111]:
```suggestion
              throw new Error('System requirements have not been met');
```

**Review by thephez — COMMENTED** [review]
@shumkov I left a few suggestions. All pretty minor stuff

File: packages/dashmate/src/listr/tasks/setup/regular/verifySystemRequirementsTaskFactory.js [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+{red ${warningsText}}
+
+    Dash Platform needs more minerals`;
+            // TODO: Write some text here
+
+            const proceed = await task.prompt({
+              type: 'toggle',
+              header,
+              message: 'Are you sure you want to proceed?',
```
Comment (thephez) [inline packages/dashmate/src/listr/tasks/setup/regular/verifySystemRequirementsTaskFactory.js:104]:
```suggestion
              message: 'Are you sure you want to proceed given the risk of non-payment?',
```

File: packages/dashmate/src/listr/tasks/setup/regular/verifySystemRequirementsTaskFactory.js (hunk 3 above)
Comment (thephez) [reply packages/dashmate/src/listr/tasks/setup/regular/verifySystemRequirementsTaskFactory.js:99]: @shumkov please see this revised wording

**Review by pshenmic — DISMISSED** [review]

**Pull request merged**